src/Microsoft.Azure.WebJobs.Host/Bindings/BindingBase.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Host.Protocols;
+
+namespace Microsoft.Azure.WebJobs.Host.Bindings
+{
+
+    // Helper class for implementing IBinding with the attribute resolver pattern. 
+    internal abstract class BindingBase<TAttribute> : IBinding
+        where TAttribute : Attribute
+    {
+        protected readonly AttributeCloner<TAttribute> _cloner;
+        private readonly ParameterDescriptor _param;
+
+        public BindingBase(AttributeCloner<TAttribute> cloner, ParameterDescriptor param)
+        {
+            _cloner = cloner;
+            _param = param;
+        }
+
+        public BindingBase(AttributeCloner<TAttribute> cloner, ParameterInfo parameterInfo)
+        {
+            _cloner = cloner;
+            _param = new ParameterDescriptor
+            {
+                Name = parameterInfo.Name,
+                DisplayHints = new ParameterDisplayHints
```

Comment:
Existing extensions provide all the Dashboard strings required for a good experience here, e.g. the Prompt,Description,DefaultValue so the Invoke/Replay UI is descriptive.

Curious what the UI looks like in this case. Does the Dashboard use an empty prompt in this case, or should you provide a default value?